### ISSUE TITLE
`c.req.parseBody` instead of body-parse middleware

### ISSUE THREAD

**Issue by yusukebe** [honojs/hono#362]
There is ["body-parse" middleware](https://github.com/honojs/hono/tree/master/src/middleware/body-parse). It's useful to handle Request body. But, it's not necessary to be *middleware*. We can make the same function as `parseBody()` method in Request object:

```ts
const data = c.req.parseBody()
```

This is cleaner and does not affect performance in this way. "body-parse" middleware will be obsolete.

**Comment by matthewrobb** [comment]
If parseBody implies a json parser and doesn't support arbitrary or custom parsing then perhaps it should match the response signature and just be
```ts
const data = await c.req.json()
```

**Comment by yusukebe** [comment]
Yes, I know it. But, `c.req.parseBody()` will parse not only JSON but also such form data from `application/x-www-form-urlencoded`. It means that we'll replace this middleware:

https://github.com/honojs/hono/blob/master/src/middleware/body-parse/index.ts#L11
